### PR TITLE
Minor cleanup of react_to_want_to_compute_task

### PR DESCRIPTION
**Pull request by Krigpl**
Remove repeated code and reduce nesting complexity. No functional changes here.

### CODE REVIEW

File: golem/task/tasksession.py [hunk 1]
```diff
@@ -424,26 +424,24 @@ def send_start_session_response(self, conn_id):
...
-        if msg.concent_enabled and not self.concent_service.enabled:
+        def _cannot_assign(reason):
+            logger.debug("Cannot assign task: %r", reason)
             self.send(
                 message.tasks.CannotAssignTask(
                     task_id=msg.task_id,
-                    reason=reasons.ConcentDisabled,
-                )
+                    reason=reason,
+                ),
```
Comment (Wiezzel) [inline golem/task/tasksession.py:433]: This comma hurts me. 😿 We don't expect the argument list to expand, do we?

**Review by maaktweluit — APPROVED** [review]
LGTM!

File: golem/task/tasksession.py [hunk 2]
```diff
@@ -483,7 +480,12 @@ def _react_to_want_to_compute_task(self, msg):
             node_name_id,
         )
 
-        if task_server_ok and self.task_manager.check_next_subtask(
+        if not task_server_ok:
+            _cannot_assign(reasons.NoMoreSubtasks)
+            return
+
+        ctd = None
+        if self.task_manager.check_next_subtask(
```
Comment (Wiezzel) [inline golem/task/tasksession.py]: Maybe like this?
```python
if not self.task_manager.check_next_subtask(...):
     _cannot_assign(reasons.NoMoreSubtasks)
     return
```

**Review by Wiezzel — APPROVED** [review]
Good, just 2 suggestions.

File: golem/task/tasksession.py (hunk 1 above)
Comment (Krigpl) [reply golem/task/tasksession.py:433]: You never know. I'm not attached to it but I like to keep consistent style everywhere.

File: golem/task/tasksession.py (hunk 2 above)
Comment (Krigpl) [reply golem/task/tasksession.py]: Will do.

File: golem/task/tasksession.py (hunk 1 above)
Comment (Wiezzel) [reply golem/task/tasksession.py:433]: 😢

**Comment by shadeofblue** [comment]
<3

**Pull request merged**